src/util/hash.rs

```diff
@@ -108,59 +108,59 @@ impl Sha256dEncoder {
 }
 
 impl SimpleEncoder for Sha256dEncoder {
```

Comment:
We should impl `Write` for this rather than `SimpleEncoder`, then the default impl for all `Write`s can take over.

Reply:
Addressed in: 4f76787dcf6c1a673c08bed7bdcb9ed2fe5864b3